API/Controllers/AccountController.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using VL.Shared.Interfaces;
+using VL.Shared.Model;
+
+namespace API.Controllers
+{
+    [Route("api/account")]
+    [ApiController]
+    public class AccountController : ControllerBase
+    {
+        private readonly ILogger<AccountController> _logger;
+        private readonly IAccountService _accountService;
+
+        public AccountController(
+            ILogger<AccountController> logger,
+            IAccountService accountService)
+        {
+            _logger = logger;
+            _accountService = accountService;
+        }
+
+        public static IdentityUser user = new IdentityUser();
+
+        [HttpPost]
+        [Route("register")]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> Register([FromBody] UserDto userDto)
+        {
+            _logger.LogInformation($"Registration Attempt for {userDto.Email}");
+
+            var result = await _accountService.RegisterAsync(userDto);
+
+            if (!result.Succeeded)
+            {
+                return BadRequest();
+            }
+
+            await Login(userDto);
```

Comment:
Don't call a Controller method from another Controller method

Implement the calls and responses you want in the controller and rework the logic as needed to support your design